scipy/signal/_fir_filter_design.py

```diff
@@ -1293,3 +1293,67 @@ def minimum_phase(h, method='homomorphic', n_fft=None):
         h_minimum = h_temp.real
     n_out = n_half + len(h) % 2
     return h_minimum[:n_out]
+
+
+def fwind1(hsize, window, fc, fs=2):
+    """
+    2D FIR filter design using the window method.
+
+    This function computes the coefficients of a 2D finite impulse response
+    filter. The filter is separable with linear phase; it will be designed
+    as a product of two 1D filters with dimensions defined by `hsize`.
+
+    Parameters
+    ----------
+    hsize : tuple or list of length 2
+        Lengths of the filter in each dimension. `hsize[0]` specifies the
+        number of coefficients in the row direction and `hsize[1]` specifies
+        the number of coefficients in the column direction.
+    window : tuple or list of length 2
+        Desired window to use for each 1D filter. Each element should be
+        a string or tuple of string and parameter values. See
+        `scipy.signal.get_window` for a list of windows and required
+        parameters.
+    fc : float or 1-D array_like
+        Cutoff frequency of filter (expressed in the same units as `fs`).
+    fs : float, optional
+        The sampling frequency of the signal. Default is 2.
+
+    Returns
+    -------
+    filter_2d : (hsize[0], hsize[1]) ndarray
+        Coefficients of 2D FIR filter.
+
+    Raises
+    ------
+    ValueError
+        If `hsize` and `window` are not 2-element tuples or lists.
+
```

Comment:
Here, it would be also interesting to learn about the cases when `firwin` raises an exception.

Reply:
done